planner/core/logical_plan_builder.go

```diff
@@ -2251,7 +2268,17 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName) (L
 		return nil, ErrPartitionClauseOnNonpartitioned
 	}
 
-	possiblePaths, err := getPossibleAccessPaths(tn.IndexHints, tableInfo)
+	// Extract comment-style index hint like /*+ INDEX(t, idx1, idx2) */.
+	var hintFromComment []*ast.IndexHint
+	if hints := b.TableHints(); hints != nil {
+		for _, hint := range hints.indexHintList {
+			// Hints from comments can match both original table name and table alias.
+			if hint.tblName == tn.Name || hint.tblName == asName {
+				hintFromComment = append(hintFromComment, hint.indexHint)
+			}
```

Comment:
What if the query is like `select /*+ INDEX(t1, idx) */ * from t1 t2 join t2 t1 on t1.a = t2.a`?